source/portable/BufferManagement/BufferAllocation_2.c

```diff
@@ -261,25 +332,9 @@ NetworkBufferDescriptor_t * pxGetNetworkBufferWithDescriptor( size_t xRequestedS
 
             if( xRequestedSizeBytesCopy > 0U )
             {
-                if( ( xRequestedSizeBytesCopy < ( size_t ) baMINIMAL_BUFFER_SIZE ) )
-                {
-                    /* ARP packets can replace application packets, so the storage must be
-                     * at least large enough to hold an ARP. */
-                    xRequestedSizeBytesCopy = baMINIMAL_BUFFER_SIZE;
-                }
-
-                /* Add 2 bytes to xRequestedSizeBytesCopy and round up xRequestedSizeBytesCopy
-                 * to the nearest multiple of N bytes, where N equals 'sizeof( size_t )'. */
-                xRequestedSizeBytesCopy += 2U;
-
-                if( ( xRequestedSizeBytesCopy & ( sizeof( size_t ) - 1U ) ) != 0U )
-                {
-                    xRequestedSizeBytesCopy = ( xRequestedSizeBytesCopy | ( sizeof( size_t ) - 1U ) ) + 1U;
-                }
-
                 /* Extra space is obtained so a pointer to the network buffer can
                  * be stored at the beginning of the buffer. */
-                pxReturn->pucEthernetBuffer = ( uint8_t * ) pvPortMalloc( xRequestedSizeBytesCopy + ipBUFFER_PADDING );
+                pxReturn->pucEthernetBuffer = ( uint8_t * ) pvPortMalloc( xAllocatedBytes );
```

Comment:
Hi @tony-josi-aws,

`pucGetNetworkBuffer()` allocates a buffer for `pucEthernetBuffer`
`pxGetNetworkBufferWithDescriptor()` also allocates a buffer for `pucEthernetBuffer`

Both do length checks and add the necessary bytes.

Now I wonder why do everything double and slightly different?

Why doesn't `pxGetNetworkBufferWithDescriptor()` call `pucGetNetworkBuffer()` to obtain a buffer?

Reply:
`pucGetNetworkBuffer` doesnt need to do the +2 as its only called for resizing existing buffers.